src/main/java/org/jabref/gui/importer/ImportFormats.java

```diff
@@ -67,46 +67,41 @@ public ImportAction(boolean newDatabase) {
-                fs.setSelectedExtensionFilter(allImports);
-                File f = DefaultTaskExecutor
-                        .runInJavaFXThread(() -> fs.showOpenDialog(null));
-                Optional<Path> selectedFile = Optional.ofNullable(f).map(File::toPath);
-                FileChooser.ExtensionFilter selectedExtension = fs.getSelectedExtensionFilter();
-                // Add file filter for all supported types
-
-                selectedFile.ifPresent(file -> {
-                    try {
-                        if (!Files.exists(file)) {
-                            JOptionPane.showMessageDialog(frame,
-                                    Localization.lang("File not found") + ": '" + file.getFileName() + "'.",
-                                    Localization.lang("Import"), JOptionPane.ERROR_MESSAGE);
-                            return;
-                        }
-                        Optional<Importer> format = ImportFileFilter.convert(selectedExtension, importers);
-                        ImportMenuItem importMenu = new ImportMenuItem(frame, newDatabase, format.orElse(null));
-                        importMenu.automatedImport(Collections.singletonList(file.toString()));
-                        // Set last working dir for import
-                        Globals.prefs.put(JabRefPreferences.IMPORT_WORKING_DIRECTORY, file.getParent().toString());
-                    } catch (Exception ex) {
-                        LOGGER.warn("Cannot import file", ex);
-                    }
+                DialogService dialogService = new FXDialogService();
+                DefaultTaskExecutor.runInJavaFXThread(() -> {
+                    dialogService.showFileOpenDialog(fileDialogConfiguration)
+                            .ifPresent(path -> doImport(path, importers, fileDialogConfiguration.getSelectedExtensionFilter()));
                 });
             }
+
+            private void doImport(Path file, SortedSet<Importer> importers, FileChooser.ExtensionFilter selectedExtensionFilter) {
+                try {
+                    if (!Files.exists(file)) {
+                        JOptionPane.showMessageDialog(frame,
+                                Localization.lang("File not found") + ": '" + file.getFileName() + "'.",
+                                Localization.lang("Import"), JOptionPane.ERROR_MESSAGE);
+                        return;
+                    }
+                    Optional<Importer> format = FileFilterConverter.getImporter(selectedExtensionFilter, importers);
+                    ImportMenuItem importMenu = new ImportMenuItem(frame, newDatabase, format.orElse(null));
+                    importMenu.automatedImport(Collections.singletonList(file.toString()));
+                    // Set last working dir for import
+                    Globals.prefs.put(JabRefPreferences.IMPORT_WORKING_DIRECTORY, file.getParent().toString());
+                } catch (Exception ex) {
```

Comment:
There's certainly a more specific type of exception you can catch here, isn't it?